util/symexec.cpp

```diff
@@ -13,17 +13,32 @@ using namespace std;
 namespace util {
 
 void sym_exec(State &s) {
+  BasicBlock &initBB = const_cast<BasicBlock&>(s.getInitializerBB());
   Function &f = const_cast<Function&>(s.getFn());
 
-  // add constants & inputs to State table first of all
-  for (auto &l : { f.getConstants(), f.getInputs(), f.getUndefs() }) {
+  // add constants & undefs to State table first
+  for (auto &l : { f.getConstants(), f.getUndefs() }) {
     for (const auto &v : l) {
       s.exec(v);
     }
   }
 
   s.exec(Value::voidVal);
 
+  for (auto &i : initBB.instrs()) {
+    auto val = s.exec(i);
+    auto &name = i.getName();
+
+    if (config::symexec_print_each_value && name[0] == '%')
+      cout << name << " = " << val << '\n';
+  }
+
+  s.finishInitializer();
+
+  // Add inputs
+  for (const auto &v : f.getInputs())
+    s.exec(v);
```

Comment:
why is this done after running the init program?
So alloc will run after store? This is not particularly good and it may prevent further optimizations (like propagating sizes of non-locals).

Reply:
GlobalVariables are added as constants to the function (through `current_fn->addConstant()`), so things are okay here.